ext/rugged/rugged_commit.c

```diff
@@ -647,7 +648,8 @@ static VALUE rb_git_commit_extract_signature(int argc, VALUE *argv, VALUE self)
 		git_buf_free(&signed_data);
 	}
 
-	if (error == GIT_ENOTFOUND) {
+	err = giterr_last();
+	if (error == GIT_ENOTFOUND && err->klass == GITERR_OBJECT ) {
```

Comment:
You can call `giterr_last()->klass` here, which probably looks cleaner than the extra `err` variable.